staging/src/k8s.io/client-go/tools/record/event.go

```diff
@@ -346,9 +349,13 @@ func (recorder *recorderImpl) generateEvent(object runtime.Object, annotations m
 	// when we go to shut down this broadcaster.  Just drop events if we get overloaded,
 	// and log an error if that happens (we've configured the broadcaster to drop
 	// outgoing events anyway).
-	if sent := recorder.ActionOrDrop(watch.Added, event); !sent {
+	sent, err := recorder.ActionOrDrop(watch.Added, event)
+	if !sent {
 		klog.Errorf("unable to record event: too many queued events, dropped event %#v", event)
 	}
+	if err != nil {
+		klog.Errorf("unable to record event: %v", err)
+	}
```

Comment:
I suggest switching the above to:
```
sent, err := recorder.ActionOrDrop(watch.Added, event)
if err != nil {
  ...
  return
}
if !sent {
  ...
}
```

In other words - we should first process the error (and return, since "!sent" in that case doesn't necessary too many queued events...)